gcs/src/gcs_group.cpp

```diff
@@ -2273,7 +2251,7 @@ gcs_group_fetch_pfs_info(
 
         strncpy(entries[i].host_name, node.name, WSREP_HOSTNAME_LENGTH);
 
-        strncpy(entries[i].uuid, node.id, WSREP_UUID_STR_LEN);
+        strncpy(entries[i].uuid, node.id, WSREP_UUID_STR_LEN + 1);
```

Comment:
Why? Above we used buffer lenght-1.
I see that we are not consistent with strncpy approach. Above when we copied into interval_string_ we used buffer len-1, a line above we used a buffer len-1, but here we used the whole buffer len.
I think the better approach should beusing the whole buffer len, then explicitly set last byte to zero just in case.

Reply:
All of these buffers are allocated as size WSREP_XXX + 1.  Since strncpy does not null-terminate if the max length is reached, I agree with Kamil that it would be good to explicitly null-terminate (the node.name is the one that is most likely to cause a problem).

Reply:
Updated .